components/fuse/component_output.go

```diff
@@ -64,27 +69,47 @@ func CreateGet(cfg Config) query.GetFunc {
 				continue
 			}
 
-			prev, err := state.FindEvent(ctx, cfg.Query.State.DBRO, now.Unix(), info.DeviceName)
-			if err != nil {
+			if err := metrics.SetConnectionsCongestedPercent(ctx, info.DeviceName, info.CongestedPercent, now); err != nil {
+				return nil, err
+			}
+			if err := metrics.SetConnectionsMaxBackgroundPercent(ctx, info.DeviceName, info.MaxBackgroundPercent, now); err != nil {
 				return nil, err
 			}
-			if prev == nil {
+
+			msgs := []string{}
+			if info.CongestedPercent > cfg.CongestedPercentAgainstThreshold {
+				msgs = append(msgs, fmt.Sprintf("congested percent against threshold %.2f exceeds threshold %.2f", info.CongestedPercent, cfg.CongestedPercentAgainstThreshold))
+			}
+			if info.MaxBackgroundPercent > cfg.MaxBackgroundPercentAgainstThreshold {
+				msgs = append(msgs, fmt.Sprintf("max background percent against threshold %.2f exceeds threshold %.2f", info.MaxBackgroundPercent, cfg.MaxBackgroundPercentAgainstThreshold))
+			}
+			if len(msgs) == 0 {
 				continue
 			}
 
-			if err := state.InsertEvent(ctx, cfg.Query.State.DBRW, state.Event{
-				UnixSeconds:                          now.Unix(),
-				DeviceName:                           info.DeviceName,
-				CongestedPercentAgainstThreshold:     info.CongestedPercent,
-				MaxBackgroundPercentAgainstThreshold: info.MaxBackgroundPercent,
-			}); err != nil {
-				return nil, err
+			ib, err := info.JSON()
+			if err != nil {
+				continue
+			}
+			ev := components.Event{
+				Time:    metav1.Time{Time: now.UTC()},
+				Name:    "fuse_connections",
+				Type:    common.EventTypeCritical,
+				Message: info.DeviceName + ": " + strings.Join(msgs, ", "),
+				ExtraInfo: map[string]string{
+					"data":     string(ib),
```

Comment:
current eventStore.Find cannot correctly compare ib content..but since we already have info.DeviceName and msg included in Message maybe its fine..
the rest lgtm

Reply:
```go
func compareEvent(eventA, eventB components.Event) bool {
	if len(eventA.ExtraInfo) != len(eventB.ExtraInfo) {
		return false
	}
	for key, value := range eventA.ExtraInfo {
		if val, ok := eventB.ExtraInfo[key]; !ok || val != value {
			return false
		}
	}
	return true
}
```

should be able to cover this?

Reply:
yea this only checks kv but does not unmarshal ib..